frontstage/views/surveys.py

```diff
@@ -249,6 +260,47 @@ def upload_survey(session):
         logger.info('Upload successful', party_id=party_id, case_id=case_id)
         return render_template('surveys/surveys-upload-success.html', _theme='default', upload_filename=upload_filename)
     else:
-        logger.error('Upload failed', status_code=result.status_code, party_id=party_id, case_id=case_id)
-        error_info = {'status code': result.status_code, 'text': result.text}
-        return render_template('surveys/surveys-upload-failure.html', _theme='default', error_info=error_info, case_id=case_id)
+        if ".xlsx format" in result.text:
+            error_info = "type"
+        elif "50 characters" in result.text:
+            error_info = "charLimit"
+        else:
+            logger.error('Unexpected error message returned from collection instrument',
+                         status_code=result.status_code,
+                         party_id=party_id,
+                         case_id=case_id)
+            error_info = "unexpected"
+
+        return redirect(url_for('surveys_bp.upload_failed',
+                                _external=True,
+                                _scheme=getenv('SCHEME', 'http'),
+                                case_id=case_id,
+                                error_info=error_info))
+
+
+@surveys_bp.route('/upload_failed', methods=['GET'])
+@jwt_authorization(request)
+def upload_failed(session):
+    """Logged in page for users only."""
+    party_id = session.get('party_id', 'no-party-id')
+    case_id = request.args.get('case_id', None)
+    error_info = request.args.get('error_info', None)
+    logger.error('Upload failed', party_id=party_id, case_id=case_id)
+
+    if error_info == "type":
```

Comment:
could set these when you set error_info first to avoid more if statements

Reply:
If i set error_info to the dict before hand i cant pass it as a string query arg.